lighthouse-core/lib/page-functions.js

```diff
@@ -113,18 +113,20 @@ function getElementsInDocument(selector) {
  */
 /* istanbul ignore next */
 function getOuterHTMLSnippet(element, ignoreAttrs = []) {
-  const clone = element.cloneNode();
-
-  ignoreAttrs.forEach(attribute =>{
-    clone.removeAttribute(attribute);
-  });
-
-  const reOpeningTag = /^[\s\S]*?>/;
-  const match = clone.outerHTML.match(reOpeningTag);
-
-  return (match && match[0]) || '';
+  try {
+    const clone = element.cloneNode();
+    ignoreAttrs.forEach(attribute =>{
+      clone.removeAttribute(attribute);
+    });
+    const reOpeningTag = /^[\s\S]*?>/;
+    const match = clone.outerHTML.match(reOpeningTag);
+    return (match && match[0]) || '';
+  } catch (error) {
+    return element.localName;
```

Comment:
```suggestion
    return `<${element.localName}>`;
```

Reply:
just to match the expected output of this function under normal conditions

Reply:
Thanks a lot for the feedback @patrickhulce .
You want me to remove completely the `package-lock.json` ?
I am checking the a11y tester case now, I 've never done this before :) 
But I am trying to figure out

Reply:
![image](https://user-images.githubusercontent.com/32495928/58663638-96eb0200-8335-11e9-9727-741b685361e5.png)
It seems that it pass,
but I didnt add any error-prone element inside the a11y_tester.html.

Reply:
That's expected :)

Steps:

1. Temporarily comment out your fix
1. Add an error-prone element inside a11y_tester.html
2. Run `yarn smoke a11y`, it should fail
3. Uncomment your fix
2. Run `yarn smoke a11y`, it should pass
4. Commit the changes! 🎉

Reply:
The error-prone element should be printed somewhere in the bug report right?

Reply:
my problem is that i can't find an error-prone element to add it in the smoke a11y test,
Is this performance report supposed to show me an element?
or is there another report that displays an the error element?

Reply:
hm it looks like the problematic element has no `localName` too `<undefined>` is definitely not what we want to be returning here.

Reply:
OK I've done a bit of digging and this turned out to be a bit more complicated than it appeared, sorry about that! I'll go ahead and add the test to your PR :)

Reply:
OK new plan!

Add a test to https://github.com/GoogleChrome/lighthouse/blob/767116696fdb1fc0903d369f811005a1cdae8fbb/lighthouse-core/test/lib/page-functions-test.js#L48-L52 that tests we can handle shadow root elements and elements that fail when their `cloneNode` is called :)

i.e. `element.cloneNode = () => { throw new Error('Oops!') }` and make sure it recovers

Reply:
![image](https://user-images.githubusercontent.com/32495928/58730423-c4e34b80-83f4-11e9-8c62-03bc4e085d82.png)

Reply:
you mean something like this?

Reply:
![image](https://user-images.githubusercontent.com/32495928/58730904-104a2980-83f6-11e9-9ec0-04f19342d8ab.png)

Reply:
meh either way,
I run the test with `npm run test`
and it crashes... 
well :'(

Reply:
I meant something like

```js
it('should handle dom nodes that cannot be inspected', () => {
  const element = dom.createElement('div');
  element.cloneNode = () => { throw new Erorr('oops!') };
  assert.equal(pageFunctions.getOuterHTMLSnippet(element), '<div>');
});

Reply:
@patrickhulce  I commited the test case :)